### ISSUE TITLE
feature_id out of range when converting sfm data to COLMAP format

### ISSUE THREAD

**Issue by json87** [openMVG/openMVG#2022]
Thanks for providing the cool SfM software. Recently, I have used openmvg to process spherical images. After SfM, the spherical images are first converted into cubic perspective images by using openMVG_main_openMVGSpherical2Cubic, which are then transformed into colmap format. However, I found that the feature id in points3D.txt file is out of range (greater than the feature point index of cubic images). 

![image](https://user-images.githubusercontent.com/17582213/157035070-30e5434a-d9af-43f2-b4c2-c9374bef265d.png)

**Comment by pmoulon** [comment]
Hello thank you for your feedback.
It is expected that feature ID should be invalid, since we don't have extracted features for the cubic images.

So if you want post process for MVS the dataset you can either directly use the OpenMVG2OpenMVS.

Or if you want to use colmap, you could compute feature for the cubic scene (launch compute features, and compute matches) and then triangulate the tracks by using main_SfM with using existing poses. There you will have valid feature ids.

**Comment by json87** [comment]
@pmoulon thanks for your reply. Recently, I have debugged the related code and found that the feature id is just a index for accessing features. Besides, the relationship of data in images.txt and points3d.txt is built through the landmark.obs. Thus, I have revised the code as follows, and the converted results seem ok. In a word, I just use the index of a feature in cubic images as its feature id in points3d.txt

![image](https://user-images.githubusercontent.com/17582213/157356756-89ba9d3f-9ea8-434e-af3e-bd8430e5315f.png)
![image](https://user-images.githubusercontent.com/17582213/157356924-ae2fab16-1f26-4fd0-b010-ae3a7c40778a.png)
